In [4]:
import pandas as pd
from sklearn.cluster import KMeans
import matplotlib.pyplot as plt

# Load your data
data = pd.read_csv('finaluhivalues.csv')  

# Select relevant features
features = data[['Building Density Percentage', 'Green Area Percentage', 'Water Bodies Percentage', 'UHI Effect']]

# Apply KMeans clustering to find natural groupings
kmeans = KMeans(n_clusters=3, random_state=42)
clusters = kmeans.fit_predict(features)

# Add cluster labels to the data
data['Cluster'] = clusters



# Analyze clusters to set thresholds
for cluster in range(3):
    cluster_data = data[data['Cluster'] == cluster]
    print(f"Cluster {cluster}:")
    print(cluster_data.describe())


Cluster 0:
       Water Bodies Percentage  Green Area Percentage  \
count                 8.000000               8.000000   
mean                 30.260000               0.525000   
std                   8.422322               0.476595   
min                  19.170000               0.020000   
25%                  22.740000               0.137500   
50%                  32.315000               0.365000   
75%                  36.632500               0.885000   
max                  40.150000               1.280000   

       Building Density Percentage  UHI Effect  Cluster  
count                     8.000000    8.000000      8.0  
mean                     69.313750    0.824784      0.0  
std                       8.546973    0.462071      0.0  
min                      59.480000    0.305240      0.0  
25%                      63.040000    0.424570      0.0  
50%                      67.205000    0.871593      0.0  
75%                      77.010000    1.015559      0.0  
max        

## Cluster 0 Recommendations
- Water Bodies Percentage: Moderately high, with a mean of 30.26% and a range from 19.17% to 40.15%.
- Green Area Percentage: Relatively low, with a mean of 0.53% and a range from 0.02% to 1.28%.
- Building Density Percentage: High, with a mean of 69.31% and a range from 59.48% to 80.76%.
- UHI Index: Positive, with a mean of 0.82 and a range from 0.30 to 1.70, indicating a higher Urban Heat Island (UHI) effect.
### Conclusion for Cluster 0:
    Areas in this cluster tend to have high building density and low green area coverage, contributing to a higher UHI effect.

### Recommendations:

1) Increase Green Spaces: Create parks, gardens, and green rooftops to enhance vegetation coverage, which can help cool the area.
2) Enhance Water Bodies: Expand existing water bodies or create new ones, such as ponds or artificial lakes, to improve cooling effects.
3) Use Cool Roofs and Pavements: Implement reflective or green roofing materials and cool pavements to reduce surface temperatures.
4) Urban Tree Canopy: Increase tree planting initiatives to provide shade and reduce heat absorption in urban areas.
5) Promote Sustainable Architecture: Encourage building designs that incorporate natural ventilation, insulation, and green technologies to minimize heat retention.

## Cluster 1 Recommendations
- Water Bodies Percentage: High, with a mean of 59.44% and a range from 48.71% to 64.67%.
- Green Area Percentage: Moderately high, with a mean of 0.71% and a range from 0.01% to 1.55%.
- Building Density Percentage: Low, with a mean of 39.96% and a range from 34.12% to 51.35%.
- UHI Index: Negative, with a mean of -0.17 and a range from -0.35 to 0.04, indicating a lower Urban Heat Island (UHI) effect.

### Conclusion for Cluster 1: 
    This cluster exhibits lower building density and higher water body coverage, resulting in a reduced UHI effect.

### Recommendations:

1) Maintain Water Bodies: Ensure that existing water bodies are well-maintained and not subject to pollution to preserve their cooling effects.
2) Further Develop Green Areas: Create additional parks or green corridors that connect existing green spaces to enhance biodiversity and cooling.
3) Promote Sustainable Land Use: Implement zoning regulations that encourage lower building densities and maintain open spaces.
4) Enhance Urban Forestry Programs: Support initiatives for planting trees in urban areas, focusing on species that thrive in local climates.
5) Engage Community Awareness: Educate residents about the benefits of maintaining green areas and water bodies for mitigating heat effects.

## Cluster 2 Recommendations
- Water Bodies Percentage: Low, with a mean of 22.39% and a range from 21.37% to 23.40%.
- Green Area Percentage: Very low, with a mean of 26.40% and a range from 23.34% to 29.46%.
- Building Density Percentage: Moderate, with a mean of 51.38% and a range from 47.42% to 55.33%.
- UHI Index: Negative, with a mean of -0.38 and a range from -0.45 to -0.30, indicating a lower Urban Heat Island (UHI) effect.
### Conclusion for Cluster 2: 
    This cluster has low water body and green area percentages, yet exhibits a lower UHI effect.

### Recommendations:

1) Increase Water Features: Introduce more water features, such as fountains or small ponds, to increase local humidity and cooling.
2) Promote Green Infrastructure: Encourage the development of vertical gardens, green roofs, and community gardens to enhance vegetation.
3) Reduce Impervious Surfaces: Minimize the use of concrete and asphalt by implementing permeable paving solutions in urban planning.
4) Adopt Eco-Friendly Policies: Promote policies that favor environmentally sustainable practices in urban development.
5) Monitoring and Assessment: Regularly assess the UHI effect and adjust strategies based on the effectiveness of implemented measures.

--------------------

In [5]:
import pandas as pd

data = pd.read_csv('finaluhivalues.csv') 



In [6]:
from sklearn.cluster import KMeans
features = ['Building Density Percentage', 'Green Area Percentage', 'Water Bodies Percentage', 'UHI Effect']

X = data[features]

kmeans = KMeans(n_clusters=3, random_state=42)
kmeans.fit(X)  
data['Cluster'] = kmeans.predict(X) 


In [10]:
def generate_cluster_recommendations(cluster):
    recommendations = []

    if cluster == 0:
        recommendations.append("1. Increase Green Spaces: Create parks, gardens, and green rooftops to enhance vegetation coverage, which can help cool the area.")
        recommendations.append("2. Enhance Water Bodies: Expand existing water bodies or create new ones, such as ponds or artificial lakes, to improve cooling effects.")
        recommendations.append("3. Use Cool Roofs and Pavements: Implement reflective or green roofing materials and cool pavements to reduce surface temperatures.")
        recommendations.append("4. Urban Tree Canopy: Increase tree planting initiatives to provide shade and reduce heat absorption in urban areas.")
        recommendations.append("5. Promote Sustainable Architecture: Encourage building designs that incorporate natural ventilation, insulation, and green technologies to minimize heat retention.")

    elif cluster == 1:
        recommendations.append("1. Maintain Water Bodies: Ensure that existing water bodies are well-maintained and not subject to pollution to preserve their cooling effects.")
        recommendations.append("2. Further Develop Green Areas: Create additional parks or green corridors that connect existing green spaces to enhance biodiversity and cooling.")
        recommendations.append("3. Promote Sustainable Land Use: Implement zoning regulations that encourage lower building densities and maintain open spaces.")
        recommendations.append("4. Enhance Urban Forestry Programs: Support initiatives for planting trees in urban areas, focusing on species that thrive in local climates.")

    elif cluster == 2:
        recommendations.append("1. Increase Green Spaces: Expand urban green areas by creating new parks, planting additional trees, and developing green belts.")
        recommendations.append("2. Expand Water Bodies: Introduce more water features and increase the percentage of water bodies in urban areas to enhance cooling effects.")
        recommendations.append("3. Reduce Impervious Surfaces: Minimize the use of concrete and asphalt by implementing permeable paving solutions in urban planning.")
        recommendations.append("4. Encourage Building Modifications: Upgrade existing buildings with energy-efficient materials and design modifications to minimize heat generation.")
        recommendations.append("5. Promote Urban Greening Projects: Support and invest in urban greening initiatives to boost vegetation and improve air quality.")

    return recommendations

data['Recommendations'] = data['Cluster'].apply(generate_cluster_recommendations)

for _, row in data.iterrows():
    print(f"Location: {row['zone']}")
    print(f"Cluster: {row['Cluster']}")
    print("Recommendations:")
    for rec in row['Recommendations']:
        print(f"  {rec}")
    print()
output_data = data[['zone', 'Cluster', 'Recommendations']]

output_data.to_csv('solutionforuhinews.csv', index=False)


Location: Adyar
Cluster: 2
Recommendations:
  1. Increase Green Spaces: Expand urban green areas by creating new parks, planting additional trees, and developing green belts.
  2. Expand Water Bodies: Introduce more water features and increase the percentage of water bodies in urban areas to enhance cooling effects.
  3. Reduce Impervious Surfaces: Minimize the use of concrete and asphalt by implementing permeable paving solutions in urban planning.
  4. Encourage Building Modifications: Upgrade existing buildings with energy-efficient materials and design modifications to minimize heat generation.
  5. Promote Urban Greening Projects: Support and invest in urban greening initiatives to boost vegetation and improve air quality.

Location: Anna Nagar
Cluster: 0
Recommendations:
  1. Increase Green Spaces: Create parks, gardens, and green rooftops to enhance vegetation coverage, which can help cool the area.
  2. Enhance Water Bodies: Expand existing water bodies or create new ones, such